src/analysis.zig

```diff
@@ -1140,6 +1140,33 @@ fn resolveTypeOfNodeUncached(analyser: *Analyser, node_handle: NodeWithHandle) e
                 .handle = handle,
             };
         },
+        .block,
+        .block_semicolon,
+        .block_two,
+        .block_two_semicolon,
+        => {
+            const first_token = tree.firstToken(node);
+            if (token_tags[first_token] != .identifier) return null;
+
+            const block_label = tree.tokenSlice(first_token);
+
+            var buffer: [2]Ast.Node.Index = undefined;
+            const statements = ast.blockStatements(tree, node, &buffer).?;
+
+            for (statements) |child_idx| {
+                // TODO: Recursively find matching `break :label` (e.g. inside `if`)
```

Comment:
If you wish to follow up and resolve this TODO, helper functions like [`iterateChildrenRecursive`](https://github.com/zigtools/zls/blob/a16fb19797d627a6cd500065aacee098c624a66c/src/ast.zig#L1545) may help you traverse the ast while searching for break statements.